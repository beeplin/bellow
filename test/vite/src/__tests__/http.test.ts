describe('http run', () => {
  it('works for vite', async () => {
    await page.goto('http://localhost:3000')
    await wait(1000)

    await expect(page.$eval('#app', (el) => el.textContent)).resolves.toMatchSnapshot()
  })

  it('works for vite build', async () => {
    await page.goto('http://localhost:4000')
    await wait(1000)

    await expect(page.$eval('#app', (el) => el.textContent)).resolves.toMatchSnapshot()
  })
})

async function wait(ms: number) {
  return new Promise((resolve) => {
    setTimeout(resolve, ms)
  })
}
