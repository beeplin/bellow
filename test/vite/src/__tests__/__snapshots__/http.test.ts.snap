// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`http run works for vite 1`] = `
"[
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo\\",
        \\"id\\": \\"1001\\",
        \\"input\\": \\"A\\"
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo\\",
        \\"id\\": \\"1001\\",
        \\"input\\": \\"B\\"
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"change\\",
        \\"id\\": \\"1002\\"
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo\\",
        \\"id\\": \\"1002\\",
        \\"input\\": \\"C\\"
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo\\",
        \\"id\\": \\"1002\\",
        \\"input\\": \\"D\\"
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo1\\",
        \\"id\\": \\"1002\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1001\\",
            \\"input\\": \\"E\\"
        }
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo1\\",
        \\"id\\": \\"1002\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1001\\",
            \\"input\\": \\"F\\"
        }
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"change1\\",
        \\"id\\": \\"1002\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"change\\",
            \\"id\\": \\"1002\\"
        }
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo1\\",
        \\"id\\": \\"1002\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1002\\",
            \\"input\\": \\"G\\"
        }
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo1\\",
        \\"id\\": \\"1002\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1002\\",
            \\"input\\": \\"H\\"
        }
    },
    [
        {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1003\\",
            \\"input\\": \\"I\\"
        },
        {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"change\\",
            \\"id\\": \\"1004\\"
        },
        {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1004\\",
            \\"input\\": \\"I\\"
        }
    ],
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo1\\",
        \\"id\\": \\"1002\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1002\\",
            \\"input\\": \\"J\\"
        }
    }
]"
`;

exports[`http run works for vite build 1`] = `
"[
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo\\",
        \\"id\\": \\"1002\\",
        \\"input\\": \\"A\\"
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo\\",
        \\"id\\": \\"1002\\",
        \\"input\\": \\"B\\"
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"change\\",
        \\"id\\": \\"1003\\"
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo\\",
        \\"id\\": \\"1003\\",
        \\"input\\": \\"C\\"
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo\\",
        \\"id\\": \\"1003\\",
        \\"input\\": \\"D\\"
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo1\\",
        \\"id\\": \\"1003\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1002\\",
            \\"input\\": \\"E\\"
        }
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo1\\",
        \\"id\\": \\"1003\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1002\\",
            \\"input\\": \\"F\\"
        }
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"change1\\",
        \\"id\\": \\"1003\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"change\\",
            \\"id\\": \\"1005\\"
        }
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo1\\",
        \\"id\\": \\"1003\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1005\\",
            \\"input\\": \\"G\\"
        }
    },
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo1\\",
        \\"id\\": \\"1003\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1005\\",
            \\"input\\": \\"H\\"
        }
    },
    [
        {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1004\\",
            \\"input\\": \\"I\\"
        },
        {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"change\\",
            \\"id\\": \\"1006\\"
        },
        {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1006\\",
            \\"input\\": \\"I\\"
        }
    ],
    {
        \\"server\\": \\"server0\\",
        \\"method\\": \\"echo1\\",
        \\"id\\": \\"1003\\",
        \\"res\\": {
            \\"server\\": \\"server1\\",
            \\"method\\": \\"echo\\",
            \\"id\\": \\"1005\\",
            \\"input\\": \\"J\\"
        }
    }
]"
`;
